### ISSUE TITLE
Testnet node crashes on contract call for non-existent contract

### ISSUE THREAD

**Issue by hstove** [blockstack/stacks-blockchain#1406]
To repro:

- Run a testnet node
- Broadcast a contract-call TX (here's on in hex: `800000000004004a4576a2dd3b287a86a52a66b53cc95f281690ef0000000000000000000000000000000001013dd2ddc3ad5b09d6a20618713b99ef5a1d176c741e8294cb1330f75289a6452e0c5286a424084e9a4649e023ec555393c2e85d8ec70a089f098e545aeffd67e1030200000000021a85a37f278ef8ea7e317cce78921aab75cadaeaea0673747265616d067361792d686900000000`)
- See that the node panics and crashes

```
WARN [1586404532.250] [src/chainstate/stacks/db/transactions.rs:420] Invalid contract-call transaction 3084bd6e7f78e5d41ff3a3cffbdd88e516b38bb558dc490237acdf5d8a9939d2: Interpreter(Unchecked(NoSuchContract("ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ.stream")))
ERROR [1586404532.250] [src/testnet/helium/tenure.rs:66] Failed mining transaction - Unchecked(NoSuchContract("ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ.stream"))
Burnchain block #5 (376da11fe3ab3d0eaaddb418ccb49b5426d5c2504f526f7766580f6e45984e3b) was produced with sortition #66b19d09f5d45a075f47e651635243593d8611926883425abd0e5e1fcfb10ff8
WARN [1586404532.347] [src/chainstate/stacks/db/blocks.rs:2692] Block 7f649d0bc8c78a845f57d5d39520a9e4dd902719b3e7163017c5f1bc4c5367b2 state root mismatch: expected 92eae788c8bf948bb17202f95eecb48ceb71b99209b855782d3880ffbf476166, got ffea29bb06805ac7d07e0226f36e3bdea5f2d8c5a8c627048ac4f4af4bc2673b
WARN [1586404532.360] [src/chainstate/stacks/db/blocks.rs:2962] Encountered invalid block: Block 7f649d0bc8c78a845f57d5d39520a9e4dd902719b3e7163017c5f1bc4c5367b2 state root mismatch: expected 92eae788c8bf948bb17202f95eecb48ceb71b99209b855782d3880ffbf476166, got ffea29bb06805ac7d07e0226f36e3bdea5f2d8c5a8c627048ac4f4af4bc2673b
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2791:10
stack backtrace:
   0: std::sys_common::at_exit_imp::push
   1: core::fmt::ArgumentV1::show_usize
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
   6: std::panicking::try::do_call
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::panicking::panic_bounds_check
             at src/libcore/panicking.rs:63
   9: <usize as core::slice::SliceIndex<[T]>>::index
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2791
  10: core::slice::<impl core::ops::index::Index<I> for [T]>::index
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2656
  11: <alloc::vec::Vec<T> as core::ops::index::Index<I>>::index
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/vec.rs:1883
  12: blockstack_lib::testnet::helium::node::Node::process_tenure
             at src/testnet/helium/node.rs:425
  13: blockstack_lib::testnet::helium::run_loop::RunLoop::start
             at src/testnet/helium/run_loop.rs:193
  14: blockstack_core::main
             at src/main.rs:206
  15: std::rt::lang_start::{{closure}}
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  16: std::panicking::try::do_call
  17: panic_unwind::dwarf::eh::read_encoded_pointer
  18: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
  19: std::rt::lang_start
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  20: blockstack_core::main
```

**Comment by zone117x** [comment]
Related https://github.com/blockstack/stacks-blockchain/issues/1384

**Comment by diwakergupta** [comment]
Should also be resolved by fix for #1376

**Comment by diwakergupta** [comment]
Fixed by #1415
